### ISSUE TITLE
mtu set too high

### ISSUE THREAD

**Issue by hyber-code** [rajannpatel/Pi-Hole-on-Google-Compute-Engine-Free-Tier-with-Full-Tunnel-and-Split-Tunnel-Wireguard-VPN-Configs#33]
ISSUE

some times the wireguard client sets mtu=8920 which is too high and as a result some devices does not load sites and other data properly,

i manually changed the mtu by going to following directory
```
cd /sys/class/net/wg0
nano mtu
and changing the value to 1380
```
can this be made the default value ?

it took me hours the catch the issue on one of my networks.

**Comment by hyber-code** [comment]
it has something to do with following lines in your script

```
# Read MTU value
	if [ -f "/sys/class/net/${SERVER_WG_NIC}/mtu" ]; then
		CLIENT_MTU="MTU = $(cat /sys/class/net/${SERVER_WG_NIC}/mtu)"
```

**Comment by rajannpatel** [comment]
I will look into this... for google compute engine it may make sense to set the MTU on the interface as well:

```bash
 sudo ip link set dev wg0 mtu 1360
```

What is the best way to read and set the correct MTU value on the server, and config files? Do you have any thoughts around this?

**Comment by rajannpatel** [comment]
This MTU issue seems to be resolved in the current version of Wireguard available in Ubuntu 24.04 LTS.
